src/plugins/vue.js

```diff
@@ -5,59 +5,63 @@ function hasValueOrEmptyAttribute(value) {
+    data: function() {
+      return {
+        detectedLanguage: "",
+        unknownLanguage: false
+      };
     },
-    highlighted() {
-      // no idea what language to use, return raw code
-      if (!this.autoDetect && !hljs.getLanguage(this.language)) {
-        console.warn(`The language "${this.language}" you specified could not be found.`);
-        this.unknownLanguage = true;
-        return escapeHTML(this.code);
-      }
-
-      let result;
-      if (this.autoDetect) {
-        result = hljs.highlightAuto(this.code);
-        this.detectedLanguage = result.language;
-      } else {
-        result = hljs.highlight(this.language, this.code, this.ignoreIllegals);
-        this.detectectLanguage = this.language;
+    computed: {
+      className() {
+        if (this.unknownLanguage) return "";
+  
+        return "hljs " + this.detectedLanguage;
+      },
+      highlighted() {
+        // no idea what language to use, return raw code
+        if (!this.autoDetect && !hljs.getLanguage(this.language)) {
+          console.warn(`The language "${this.language}" you specified could not be found.`);
+          this.unknownLanguage = true;
+          return escapeHTML(this.code);
+        }
+  
+        let result;
+        if (this.autoDetect) {
+          result = hljs.highlightAuto(this.code);
+          this.detectedLanguage = result.language;
+        } else {
+          result = hljs.highlight(this.language, this.code, this.ignoreIllegals);
+          this.detectectLanguage = this.language;
+        }
+        return result.value;
+      },
+      autoDetect() {
+        return !this.language || hasValueOrEmptyAttribute(this.autodetect);
+      },
+      ignoreIllegals() {
+        return true;
       }
-      return result.value;
     },
-    autoDetect() {
-      return !this.language || hasValueOrEmptyAttribute(this.autodetect);
-    },
-    ignoreIllegals() {
-      return true;
+    // this avoids needing to use a whole Vue compilation pipeline just
+    // to build Highlight.js
+    render(createElement) {
+      return createElement("pre", {}, [
+        createElement("code", {
+          class: this.className,
+          domProps: { innerHTML: this.highlighted }})
+      ]);
+    }
+    // template: `<pre><code :class="className" v-html="highlighted"></code></pre>`
+  };
+  
+  export const VuePlugin = {
+    install(Vue) {
+      Vue.component('highlightjs', Component);
     }
-  },
-  // this avoids needing to use a whole Vue compilation pipeline just
-  // to build Highlight.js
-  render(createElement) {
-    return createElement("pre", {}, [
-      createElement("code", {
-        class: this.className,
-        domProps: { innerHTML: this.highlighted }})
-    ]);
-  }
-  // template: `<pre><code :class="className" v-html="highlighted"></code></pre>`
-};
+  };
 
-export const VuePlugin = {
-  install(Vue) {
-    Vue.component('highlightjs', Component);
-  }
-};
+  return { Component, VuePlugin };
```

Comment:
What does exporting component do for us since we never import it anywhere?  I would understand the intent in a larger sense, but this file isn't distributed anywhere - it's only built inside of HLJS.

Reply:
I don't know that there is one and I'd be happy to change that.
I just did it to make my change more closely resemble the original, which did export `Component` event though it was never referenced.

Reply:
> I just did it to make my change more closely resemble the original, which did export Component event though it was never referenced.

HAHA, so it's my fault... far to easy to slap `export` next to things in the middle... it pops more when you put them all in a single place. No need to change anything.